pymarsys/contact.py

```diff
@@ -32,3 +32,290 @@ def create(self, contact, key_id=None, source_id=None):
+            'return': return_,
+        }
+
+        for param_name in ('limit', 'offset', 'excludeempty'):
+            param_value = locals()[param_name]
+            if param_value is not None:
+                params[param_name] = param_value
+
+        if field_id_dict is not None:
+            if len(field_id_dict) > 1:
+                raise ValueError('Only one field_id is allowed.')
+            (key, value), = field_id_dict.items()
+            params[key] = value
+
+        return self.connection.make_call(
+            'GET',
+            query_endpoint,
+            params=params
+        )
+
+    def get_data(self,
+                 key_id,
+                 key_values,
+                 fields=None):
+        """
+        Returns the values of specified fields for contacts. The contacts can
+        be specified by using either the internal IDs or by using another
+        column value.
+        :param key_id: Key which identifies the contacts.
+        :param key_values: Value of the keyId.
+        :param fields: Define which system fields to include in the output.
+        :return: Values of specified fields for contacts.
+        """
+        query_endpoint = '{}/{}/'.format(self.endpoint, 'getdata')
+        payload = {
+            'keyId': key_id,
+            'keyValues': key_values,
+        }
+
+        if fields:
+            payload['fields'] = fields
+
+        return self.connection.make_call(
+            'POST',
+            query_endpoint,
+            payload=payload
+        )
+
+    def get_history(self,
+                    contacts,
+                    start_date=None,
+                    end_date=None):
+        """
+        Returns a list of email campaign launch data for specified contacts,
+        can also be restricted to specified timeframe (optional).
+        :param contacts: Integer array which contains the contact IDs to
+        include.
+        :param start_date: yyyy-mm-dd formatted date string used to filter
+        emails by the date the launch was initiated.
+        :param end_date: yyyy-mm-dd formatted date string used to filter
+        emails by the date the launch completed.
+        :return: List of email campaign launch data.
+        """
+        query_endpoint = '{}/{}/'.format(self.endpoint, 'getcontacthistory')
+        payload = {
+            'contacts': contacts,
+        }
+
+        if start_date:
+            payload['startDate'] = start_date
+
+        if end_date:
+            payload['endate'] = end_date
+
+        return self.connection.make_call(
+            'POST',
+            query_endpoint,
+            payload=payload
+        )
+
+    def fetch_internal_id(self,
+                          key_field_id,
+                          key_field_value):
+        """
+        Returns the internal ID of a contact based on a specified key ID.
+        :param key_field_id: ID of the key field to use.
+        :param key_field_value: Key field value to use in the query.
```

Comment:
key field id and value are not very clear, either make the documentation very explicit, or change the name to make it more explicit. Both would be great. (this is not the only place).